### PR TITLE
fixed index highlights in toggle and keydown

### PR DESCRIPTION
**Pull request by silviuavram**
**What**:

Fixes item highlights in the case of passing `highlightedIndex`, `defaultHighlightedIndex` and `initialHighlightedIndex`. Handles both toggle and opening by keyDown and keyUp. Also has some code refactoring + unit tests for `getInputProps`. Planning to also add unit tests for `getToggleButtonProps`.

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged
      
- [ ] Added myself to contributors table

### CODE REVIEW

File: src/downshift.js [hunk 1]
```diff
@@ -561,18 +533,50 @@ class Downshift extends Component {
...
-      })
+
+      if (this.getState().isOpen) {
+        const amount = event.shiftKey ? 5 : 1
+        this.moveHighlightedIndex(amount, {
+          type: stateChangeTypes.keyDownArrowDown,
+        })
+      } else {
+        const self = this
+        this.toggleMenu({type: stateChangeTypes.keyDownArrowDown}, () => {
```
Comment (GonchuB) [inline src/downshift.js]: nice 👌

File: src/__tests__/downshift.get-input-props.js [hunk 2]
```diff
@@ -90,55 +90,177 @@ test('manages arrow up and down behavior', () => {
...
+    )
+  })
+
+  test('initialHighlightedIndex - 1', () => {
+    const initialHighlightedIndex = 3
+    const {arrowUpInput, childrenSpy} = renderDownshift({
+      // provide only highlightedIndex
+      props: {initialHighlightedIndex},
+    })
+    // ↓
```
Comment (GonchuB) [inline src/__tests__/downshift.get-input-props.js]: should be `// ↑` in this and the following

File: src/downshift.js [hunk 3]
```diff
@@ -561,18 +533,50 @@ class Downshift extends Component {
...
     },
 
     ArrowUp(event) {
       event.preventDefault()
-      const amount = event.shiftKey ? -5 : -1
-      this.moveHighlightedIndex(amount, {
-        type: stateChangeTypes.keyDownArrowUp,
-      })
+
+      if (this.getState().isOpen) {
```
Comment (GonchuB) [inline src/downshift.js]: The only difference I see between these handler and ArrowDown is the `stateChangeType` and `newIndex`. Does it make sense to move to a method and pass the correct newIndex value?

Maybe it complicates things as `newIndex` is calculated from state after the menu gets toggled. And repeating something twice could be better than a wrong abstraction.

File: src/__tests__/downshift.get-input-props.js (hunk 2 above)
Comment (silviuavram) [reply src/__tests__/downshift.get-input-props.js]: thanks, will change!

**Pull request merged**